test/gtest/ucs/test_datatype.cc

```diff
@@ -918,3 +919,33 @@ UCS_TEST_F(test_datatype, fixed_array) {
     EXPECT_EQ(initial_capacity - 1, ucs_array_available_length(&test_array));
     EXPECT_EQ(&ucs_array_elem(&test_array, 1), ucs_array_end(&test_array));
 }
+
+UCS_TEST_F(test_datatype, ptr_map) {
+    typedef std::map<ucs_ptr_map_key_t, char*> std_map_t;
+
+    const size_t N = 10 * UCS_KBYTE;
+    ucs_ptr_map_key_t ptr_key;
+    ucs_ptr_map_t     ptr_map;
+    std_map_t         std_map;
+
+    ucs_ptr_map_init(&ptr_map);
+    for (size_t i = 0; i < N; ++i) {
+        char *ptr = new char;
+        if (ucs::rand() % 2) {
+            ptr_key = ucs_ptr_map_put(&ptr_map, ptr);
+        } else {
+            ptr_key = ucs_ptr_map_put_safe(&ptr_map, ptr);
+        }
```

Comment:
bad idea to use ```rand``` in API testing: there is no robust way to reproduce issue. may be add one more layer to test ucs_ptr_map_put/ucs_ptr_map_put_safe based on parameter?

Reply:
the idea was to test put/put_safe in the same map because this case is relevant for UCP